In [1]:
from warnings import filterwarnings
filterwarnings('ignore')

Step 1: Read the dataset

In [2]:
import pandas as pd
df = pd.read_csv("Cars93.csv", na_values=['', 'NA'], keep_default_na=False)
df.head()

id Manufacturer    Model     Type  Min.Price  Price  Max.Price  MPG.city  \
0   1        Acura  Integra    Small       12.9   15.9       18.8        25   
1   2        Acura   Legend  Midsize       29.2   33.9       38.7        18   
2   3         Audi       90  Compact       25.9   29.1       32.3        20   
3   4         Audi      100  Midsize       30.8   37.7       44.6        19   
4   5          BMW     535i  Midsize       23.7   30.0       36.2        22   

   MPG.highway             AirBags  ... Passengers Length  Wheelbase  Width  \
0           31                None  ...          5    177        102     68   
1           25  Driver & Passenger  ...          5    195        115     71   
2           26         Driver only  ...          5    180        102     67   
3           26                 NaN  ...          6    193        106     70   
4           30         Driver only  ...          4    186        109     69   

   Turn.circle  Rear.seat.room Luggage.room  Weight   Origin           Make  
0           37            26.5         11.0    2705  non-USA  Acura Integra  
1           38            30.0         15.0    3560  non-USA   Acura Legend  
2           37            28.0         14.0    3375  non-USA        Audi 90  
3           37            31.0         17.0    3405  non-USA       Audi 100  
4           39            27.0         13.0    3640  non-USA       BMW 535i  

[5 rows x 28 columns]

Step 2 : Perform Basic Data Quality

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 93 entries, 0 to 92
Data columns (total 28 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   id                  93 non-null     int64  
 1   Manufacturer        93 non-null     object 
 2   Model               93 non-null     object 
 3   Type                93 non-null     object 
 4   Min.Price           93 non-null     float64
 5   Price               93 non-null     float64
 6   Max.Price           93 non-null     float64
 7   MPG.city            93 non-null     int64  
 8   MPG.highway         93 non-null     int64  
 9   AirBags             89 non-null     object 
 10  DriveTrain          93 non-null     object 
 11  Cylinders           93 non-null     object 
 12  EngineSize          93 non-null     float64
 13  Horsepower          93 non-null     int64  
 14  RPM                 93 non-null     int64  
 15  Rev.per.mile        93 non-null     int64  
 16  Man.trans.

In [4]:
##check missing values
m = df.isna().sum()
m

id                     0
Manufacturer           0
Model                  0
Type                   0
Min.Price              0
Price                  0
Max.Price              0
MPG.city               0
MPG.highway            0
AirBags                4
DriveTrain             0
Cylinders              0
EngineSize             0
Horsepower             0
RPM                    0
Rev.per.mile           0
Man.trans.avail        0
Fuel.tank.capacity     0
Passengers             0
Length                 0
Wheelbase              0
Width                  0
Turn.circle            0
Rear.seat.room         2
Luggage.room          11
Weight                 0
Origin                 0
Make                   0
dtype: int64

In [5]:
df.duplicated().sum()

0

In [6]:
## Drop duplicated entries
df = df.drop_duplicates(keep='first')

Step 3 : Seperate categorical (Text) and Continuous (Number) Features

In [7]:
df.columns

Index(['id', 'Manufacturer', 'Model', 'Type', 'Min.Price', 'Price',
       'Max.Price', 'MPG.city', 'MPG.highway', 'AirBags', 'DriveTrain',
       'Cylinders', 'EngineSize', 'Horsepower', 'RPM', 'Rev.per.mile',
       'Man.trans.avail', 'Fuel.tank.capacity', 'Passengers', 'Length',
       'Wheelbase', 'Width', 'Turn.circle', 'Rear.seat.room', 'Luggage.room',
       'Weight', 'Origin', 'Make'],
      dtype='object')

In [8]:
df.dtypes

id                      int64
Manufacturer           object
Model                  object
Type                   object
Min.Price             float64
Price                 float64
Max.Price             float64
MPG.city                int64
MPG.highway             int64
AirBags                object
DriveTrain             object
Cylinders              object
EngineSize            float64
Horsepower              int64
RPM                     int64
Rev.per.mile            int64
Man.trans.avail        object
Fuel.tank.capacity    float64
Passengers              int64
Length                  int64
Wheelbase               int64
Width                   int64
Turn.circle             int64
Rear.seat.room        float64
Luggage.room          float64
Weight                  int64
Origin                 object
Make                   object
dtype: object

In [9]:
cat = list(df.columns[df.dtypes=='object'])
cat

['Manufacturer',
 'Model',
 'Type',
 'AirBags',
 'DriveTrain',
 'Cylinders',
 'Man.trans.avail',
 'Origin',
 'Make']

In [10]:
con = list(df.columns[df.dtypes!='object'])
con

['id',
 'Min.Price',
 'Price',
 'Max.Price',
 'MPG.city',
 'MPG.highway',
 'EngineSize',
 'Horsepower',
 'RPM',
 'Rev.per.mile',
 'Fuel.tank.capacity',
 'Passengers',
 'Length',
 'Wheelbase',
 'Width',
 'Turn.circle',
 'Rear.seat.room',
 'Luggage.room',
 'Weight']

Step 4 : Replace the missing values

1. Categorical (Text) - Mode
2. Continuous (numerical) - Mean,Median

In [11]:
df['AirBags'].mode()

0    Driver only
Name: AirBags, dtype: object

In [12]:
df['AirBags'].value_counts()

AirBags
Driver only           42
None                  33
Driver & Passenger    14
Name: count, dtype: int64

In [13]:
df['AirBags'].mode()[0]

'Driver only'

In [14]:
df['Rear.seat.room'].mean()

27.82967032967033

In [15]:
df['Luggage.room'].median()

14.0

Function called as df.fillna(value)

In [16]:
def replacer(df):
    cat = list(df.columns[df.dtypes=='object'])
    for i in df.columns:
        if i in cat:
            md = df[i].mode()[0]
            df[i] = df[i].fillna(md)
        else:
            mn = df[i].mean()
            df[i] = df[i].fillna(mn)
    print("Missing Values Replaced")

In [17]:
replacer(df)

Missing Values Replaced


In [18]:
df.isna().sum()

id                    0
Manufacturer          0
Model                 0
Type                  0
Min.Price             0
Price                 0
Max.Price             0
MPG.city              0
MPG.highway           0
AirBags               0
DriveTrain            0
Cylinders             0
EngineSize            0
Horsepower            0
RPM                   0
Rev.per.mile          0
Man.trans.avail       0
Fuel.tank.capacity    0
Passengers            0
Length                0
Wheelbase             0
Width                 0
Turn.circle           0
Rear.seat.room        0
Luggage.room          0
Weight                0
Origin                0
Make                  0
dtype: int64

In [19]:
df.head()

id Manufacturer    Model     Type  Min.Price  Price  Max.Price  MPG.city  \
0   1        Acura  Integra    Small       12.9   15.9       18.8        25   
1   2        Acura   Legend  Midsize       29.2   33.9       38.7        18   
2   3         Audi       90  Compact       25.9   29.1       32.3        20   
3   4         Audi      100  Midsize       30.8   37.7       44.6        19   
4   5          BMW     535i  Midsize       23.7   30.0       36.2        22   

   MPG.highway             AirBags  ... Passengers Length  Wheelbase  Width  \
0           31                None  ...          5    177        102     68   
1           25  Driver & Passenger  ...          5    195        115     71   
2           26         Driver only  ...          5    180        102     67   
3           26         Driver only  ...          6    193        106     70   
4           30         Driver only  ...          4    186        109     69   

   Turn.circle  Rear.seat.room Luggage.room  Weight   Origin           Make  
0           37            26.5         11.0    2705  non-USA  Acura Integra  
1           38            30.0         15.0    3560  non-USA   Acura Legend  
2           37            28.0         14.0    3375  non-USA        Audi 90  
3           37            31.0         17.0    3405  non-USA       Audi 100  
4           39            27.0         13.0    3640  non-USA       BMW 535i  

[5 rows x 28 columns]

In [20]:
df.to_csv("Cars93Clean.csv", index=False)

Step 5 : Seperate X and Y (Weight)

id(Serial Number ) - Satistically insignificant

In [21]:
X = df.drop(columns=['id', 'Weight'])
Y = df[['Weight']]

In [22]:
X.head()

Manufacturer    Model     Type  Min.Price  Price  Max.Price  MPG.city  \
0        Acura  Integra    Small       12.9   15.9       18.8        25   
1        Acura   Legend  Midsize       29.2   33.9       38.7        18   
2         Audi       90  Compact       25.9   29.1       32.3        20   
3         Audi      100  Midsize       30.8   37.7       44.6        19   
4          BMW     535i  Midsize       23.7   30.0       36.2        22   

   MPG.highway             AirBags DriveTrain  ... Fuel.tank.capacity  \
0           31                None      Front  ...               13.2   
1           25  Driver & Passenger      Front  ...               18.0   
2           26         Driver only      Front  ...               16.9   
3           26         Driver only      Front  ...               21.1   
4           30         Driver only       Rear  ...               21.1   

   Passengers  Length  Wheelbase  Width Turn.circle  Rear.seat.room  \
0           5     177        102     68          37            26.5   
1           5     195        115     71          38            30.0   
2           5     180        102     67          37            28.0   
3           6     193        106     70          37            31.0   
4           4     186        109     69          39            27.0   

   Luggage.room   Origin           Make  
0          11.0  non-USA  Acura Integra  
1          15.0  non-USA   Acura Legend  
2          14.0  non-USA        Audi 90  
3          17.0  non-USA       Audi 100  
4          13.0  non-USA       BMW 535i  

[5 rows x 26 columns]

In [23]:
Y.head()

Weight
0    2705
1    3560
2    3375
3    3405
4    3640

Step 6 : Perform Preprocessing on X

In [24]:
X_cat = X.select_dtypes(include='object')
X_cat

Manufacturer    Model     Type             AirBags DriveTrain Cylinders  \
0         Acura  Integra    Small                None      Front         4   
1         Acura   Legend  Midsize  Driver & Passenger      Front         6   
2          Audi       90  Compact         Driver only      Front         6   
3          Audi      100  Midsize         Driver only      Front         6   
4           BMW     535i  Midsize         Driver only       Rear         4   
..          ...      ...      ...                 ...        ...       ...   
88   Volkswagen  Eurovan      Van                None      Front         5   
89   Volkswagen   Passat  Compact                None      Front         4   
90   Volkswagen  Corrado   Sporty                None      Front         6   
91        Volvo      240  Compact         Driver only       Rear         4   
92        Volvo      850  Midsize  Driver & Passenger      Front         5   

   Man.trans.avail   Origin                Make  
0              Yes  non-USA       Acura Integra  
1              Yes  non-USA        Acura Legend  
2              Yes  non-USA             Audi 90  
3              Yes  non-USA            Audi 100  
4              Yes  non-USA            BMW 535i  
..             ...      ...                 ...  
88             Yes  non-USA  Volkswagen Eurovan  
89             Yes  non-USA   Volkswagen Passat  
90             Yes  non-USA  Volkswagen Corrado  
91             Yes  non-USA           Volvo 240  
92             Yes  non-USA           Volvo 850  

[93 rows x 9 columns]

In [25]:
df['Cylinders'].unique()

array(['4', '6', '8', '3', 'rotary', '5'], dtype=object)

In [26]:
df['Cylinders'].value_counts()

Cylinders
4         49
6         31
8          7
3          3
5          2
rotary     1
Name: count, dtype: int64

In [27]:
X_con = X.select_dtypes(exclude='object')
X_con

Min.Price  Price  Max.Price  MPG.city  MPG.highway  EngineSize  \
0        12.9   15.9       18.8        25           31         1.8   
1        29.2   33.9       38.7        18           25         3.2   
2        25.9   29.1       32.3        20           26         2.8   
3        30.8   37.7       44.6        19           26         2.8   
4        23.7   30.0       36.2        22           30         3.5   
..        ...    ...        ...       ...          ...         ...   
88       16.6   19.7       22.7        17           21         2.5   
89       17.6   20.0       22.4        21           30         2.0   
90       22.9   23.3       23.7        18           25         2.8   
91       21.8   22.7       23.5        21           28         2.3   
92       24.8   26.7       28.5        20           28         2.4   

    Horsepower   RPM  Rev.per.mile  Fuel.tank.capacity  Passengers  Length  \
0          140  6300          2890                13.2           5     177   
1          200  5500          2335                18.0           5     195   
2          172  5500          2280                16.9           5     180   
3          172  5500          2535                21.1           6     193   
4          208  5700          2545                21.1           4     186   
..         ...   ...           ...                 ...         ...     ...   
88         109  4500          2915                21.1           7     187   
89         134  5800          2685                18.5           5     180   
90         178  5800          2385                18.5           4     159   
91         114  5400          2215                15.8           5     190   
92         168  6200          2310                19.3           5     184   

    Wheelbase  Width  Turn.circle  Rear.seat.room  Luggage.room  
0         102     68           37            26.5     11.000000  
1         115     71           38            30.0     15.000000  
2         102     67           37            28.0     14.000000  
3         106     70           37            31.0     17.000000  
4         109     69           39            27.0     13.000000  
..        ...    ...          ...             ...           ...  
88        115     72           38            34.0     13.890244  
89        103     67           35            31.5     14.000000  
90         97     66           36            26.0     15.000000  
91        104     67           37            29.5     14.000000  
92        105     69           38            30.0     15.000000  

[93 rows x 17 columns]

Apply on hot encoding on categorical features

In [28]:
from sklearn.preprocessing import OneHotEncoder
ohe = OneHotEncoder(handle_unknown='ignore',
                    sparse_output=False).set_output(transform='pandas')
X_cat_pre = ohe.fit_transform(X_cat)
X_cat_pre.head()

Manufacturer_Acura  Manufacturer_Audi  Manufacturer_BMW  \
0                 1.0                0.0               0.0   
1                 1.0                0.0               0.0   
2                 0.0                1.0               0.0   
3                 0.0                1.0               0.0   
4                 0.0                0.0               1.0   

   Manufacturer_Buick  Manufacturer_Cadillac  Manufacturer_Chevrolet  \
0                 0.0                    0.0                     0.0   
1                 0.0                    0.0                     0.0   
2                 0.0                    0.0                     0.0   
3                 0.0                    0.0                     0.0   
4                 0.0                    0.0                     0.0   

   Manufacturer_Chrylser  Manufacturer_Chrysler  Manufacturer_Dodge  \
0                    0.0                    0.0                 0.0   
1                    0.0                    0.0                 0.0   
2                    0.0                    0.0                 0.0   
3                    0.0                    0.0                 0.0   
4                    0.0                    0.0                 0.0   

   Manufacturer_Eagle  ...  Make_Toyota Camry  Make_Toyota Celica  \
0                 0.0  ...                0.0                 0.0   
1                 0.0  ...                0.0                 0.0   
2                 0.0  ...                0.0                 0.0   
3                 0.0  ...                0.0                 0.0   
4                 0.0  ...                0.0                 0.0   

   Make_Toyota Previa  Make_Toyota Tercel  Make_Volkswagen Corrado  \
0                 0.0                 0.0                      0.0   
1                 0.0                 0.0                      0.0   
2                 0.0                 0.0                      0.0   
3                 0.0                 0.0                      0.0   
4                 0.0                 0.0                      0.0   

   Make_Volkswagen Eurovan  Make_Volkswagen Fox  Make_Volkswagen Passat  \
0                      0.0                  0.0                     0.0   
1                      0.0                  0.0                     0.0   
2                      0.0                  0.0                     0.0   
3                      0.0                  0.0                     0.0   
4                      0.0                  0.0                     0.0   

   Make_Volvo 240  Make_Volvo 850  
0             0.0             0.0  
1             0.0             0.0  
2             0.0             0.0  
3             0.0             0.0  
4             0.0             0.0  

[5 rows x 240 columns]

In [29]:
X_cat_pre.to_csv("ohe.csv", index=False)

Apply standard scaler on numerical features

In [30]:
from sklearn.preprocessing import StandardScaler

In [31]:
scaler = StandardScaler().set_output(transform='pandas')
X_con_pre = scaler.fit_transform(X_con)
X_con_pre.head()

Min.Price     Price  Max.Price  MPG.city  MPG.highway  EngineSize  \
0  -0.485787 -0.375720  -0.282465  0.471312     0.360925   -0.841022   
1   1.388017  1.497844   1.531409 -0.781032    -0.770514    0.515869   
2   1.008658  0.998227   0.948052 -0.423219    -0.581941    0.128186   
3   1.571949  1.893374   2.069191 -0.602126    -0.581941    0.128186   
4   0.755752  1.091905   1.303535 -0.065407     0.172352    0.806631   

   Horsepower       RPM  Rev.per.mile  Fuel.tank.capacity  Passengers  \
0   -0.073484  1.717489      1.129530           -1.062184   -0.083243   
1    1.078322  0.369586      0.005661            0.409445   -0.083243   
2    0.540813  0.369586     -0.105713            0.072197   -0.083243   
3    0.540813  0.369586      0.410659            1.359872    0.884457   
4    1.231897  0.706562      0.430909            1.359872   -1.050944   

     Length  Wheelbase     Width  Turn.circle  Rear.seat.room  Luggage.room  
0 -0.427186  -0.286932 -0.366184    -0.610436       -0.452197     -1.033015  
1  0.812171   1.629649  0.431983    -0.298510        0.738090      0.396643  
2 -0.220626  -0.286932 -0.632239    -0.610436        0.057926      0.039228  
3  0.674465   0.302785  0.165927    -0.610436        1.078172      1.111472  
4  0.192493   0.745073 -0.100128     0.013416       -0.282156     -0.318186

In [32]:
X_con_pre.shape

(93, 17)

Combine both categorical and continous preprocessed data

In [33]:
X_pre = X_con_pre.join(X_cat_pre)
X_pre.head()

Min.Price     Price  Max.Price  MPG.city  MPG.highway  EngineSize  \
0  -0.485787 -0.375720  -0.282465  0.471312     0.360925   -0.841022   
1   1.388017  1.497844   1.531409 -0.781032    -0.770514    0.515869   
2   1.008658  0.998227   0.948052 -0.423219    -0.581941    0.128186   
3   1.571949  1.893374   2.069191 -0.602126    -0.581941    0.128186   
4   0.755752  1.091905   1.303535 -0.065407     0.172352    0.806631   

   Horsepower       RPM  Rev.per.mile  Fuel.tank.capacity  ...  \
0   -0.073484  1.717489      1.129530           -1.062184  ...   
1    1.078322  0.369586      0.005661            0.409445  ...   
2    0.540813  0.369586     -0.105713            0.072197  ...   
3    0.540813  0.369586      0.410659            1.359872  ...   
4    1.231897  0.706562      0.430909            1.359872  ...   

   Make_Toyota Camry  Make_Toyota Celica  Make_Toyota Previa  \
0                0.0                 0.0                 0.0   
1                0.0                 0.0                 0.0   
2                0.0                 0.0                 0.0   
3                0.0                 0.0                 0.0   
4                0.0                 0.0                 0.0   

   Make_Toyota Tercel  Make_Volkswagen Corrado  Make_Volkswagen Eurovan  \
0                 0.0                      0.0                      0.0   
1                 0.0                      0.0                      0.0   
2                 0.0                      0.0                      0.0   
3                 0.0                      0.0                      0.0   
4                 0.0                      0.0                      0.0   

   Make_Volkswagen Fox  Make_Volkswagen Passat  Make_Volvo 240  Make_Volvo 850  
0                  0.0                     0.0             0.0             0.0  
1                  0.0                     0.0             0.0             0.0  
2                  0.0                     0.0             0.0             0.0  
3                  0.0                     0.0             0.0             0.0  
4                  0.0                     0.0             0.0             0.0  

[5 rows x 257 columns]

In [34]:
X_pre.to_csv("pre.csv", index=False)

### Apply Train Test Split
1.80% of data will be train : Will always passed to model

2.20% of data will be test : These values are hidden from model




###Random state acts like seed for random 80%, 20% split,for a given random state split will always be same

Test data split should be between 20% to 30%

In [36]:
from sklearn.model_selection import train_test_split
xtrain, xtest, ytrain, ytest = train_test_split(X_pre, Y, test_size=0.2, random_state=21)

In [37]:
xtrain.head()

Min.Price     Price  Max.Price  MPG.city  MPG.highway  EngineSize  \
57   1.365026  1.289670   1.185041 -0.423219    -0.016221   -0.356418   
31  -1.003095 -0.979424  -0.911397  0.113500     0.172352   -0.841022   
62   0.606307  0.685966   0.729294 -0.781032    -0.959087    0.322027   
29   0.043016 -0.021825  -0.063707 -0.423219    -0.204794    0.806631   
51   1.985795  1.726835   1.449374 -0.781032    -0.581941    1.872759   

    Horsepower       RPM  Rev.per.mile  Fuel.tank.capacity  ...  \
57   -0.265452 -0.304365      0.187910           -0.663618  ...   
31   -0.323043  2.054464      0.157535           -1.062184  ...   
62    1.116716  1.212025     -0.247462            0.716035  ...   
29    1.347077  0.875050     -0.713210            0.409445  ...   
51    1.270290 -1.146804     -0.996708            1.022624  ...   

    Make_Toyota Camry  Make_Toyota Celica  Make_Toyota Previa  \
57                0.0                 0.0                 0.0   
31                0.0                 0.0                 0.0   
62                0.0                 0.0                 0.0   
29                0.0                 0.0                 0.0   
51                0.0                 0.0                 0.0   

    Make_Toyota Tercel  Make_Volkswagen Corrado  Make_Volkswagen Eurovan  \
57                 0.0                      0.0                      0.0   
31                 0.0                      0.0                      0.0   
62                 0.0                      0.0                      0.0   
29                 0.0                      0.0                      0.0   
51                 0.0                      0.0                      0.0   

    Make_Volkswagen Fox  Make_Volkswagen Passat  Make_Volvo 240  \
57                  0.0                     0.0             0.0   
31                  0.0                     0.0             0.0   
62                  0.0                     0.0             0.0   
29                  0.0                     0.0             0.0   
51                  0.0                     0.0             0.0   

    Make_Volvo 850  
57             0.0  
31             0.0  
62             0.0  
29             0.0  
51             0.0  

[5 rows x 257 columns]

In [38]:
ytrain.head()

Weight
57    2920
31    2530
62    3730
29    3490
51    4055

In [39]:
xtest.head()

Min.Price     Price  Max.Price  MPG.city  MPG.highway  EngineSize  \
23  -1.003095 -0.854520  -0.701753  0.113500    -0.016221   -0.453339   
86   0.203957  0.332071   0.428500 -0.781032    -1.336233   -0.259498   
91   0.537333  0.332071   0.145937 -0.244313    -0.204794   -0.356418   
21   1.422504  1.039862   0.692834 -0.423219    -0.581941    0.612789   
17   0.100495 -0.073868  -0.209546 -0.959938    -0.581941    2.260442   

    Horsepower       RPM  Rev.per.mile  Fuel.tank.capacity  ...  \
23   -0.975733 -0.809828      0.532158           -0.816912  ...   
86   -0.111878 -0.472853      0.370159            0.961306  ...   
91   -0.572601  0.201098     -0.237337           -0.265051  ...   
21    0.060893 -0.809828     -1.108083           -0.203734  ...   
17    0.502419 -1.820755     -1.988953            1.942392  ...   

    Make_Toyota Camry  Make_Toyota Celica  Make_Toyota Previa  \
23                0.0                 0.0                 0.0   
86                0.0                 0.0                 1.0   
91                0.0                 0.0                 0.0   
21                0.0                 0.0                 0.0   
17                0.0                 0.0                 0.0   

    Make_Toyota Tercel  Make_Volkswagen Corrado  Make_Volkswagen Eurovan  \
23                 0.0                      0.0                      0.0   
86                 0.0                      0.0                      0.0   
91                 0.0                      0.0                      0.0   
21                 0.0                      0.0                      0.0   
17                 0.0                      0.0                      0.0   

    Make_Volkswagen Fox  Make_Volkswagen Passat  Make_Volvo 240  \
23                  0.0                     0.0             0.0   
86                  0.0                     0.0             0.0   
91                  0.0                     0.0             1.0   
21                  0.0                     0.0             0.0   
17                  0.0                     0.0             0.0   

    Make_Volvo 850  
23             0.0  
86             0.0  
91             0.0  
21             0.0  
17             0.0  

[5 rows x 257 columns]

In [40]:
ytest.head()

Weight
23    2670
86    3785
91    2985
21    3570
17    3910

In [41]:
xtrain.shape

(74, 257)

In [42]:
xtest.shape

(19, 257)

### Preprocessing is completed
### Build the model

In [43]:
from sklearn.linear_model import LinearRegression
model = LinearRegression()
model.fit(xtrain, ytrain)

LinearRegression()

### Evaluate the model

In [44]:
# Get r2 score for train
model.score(xtrain, ytrain)

1.0

In [45]:
# Get r2 score for test
model.score(xtest, ytest)

0.9330573934795234

### Predict the result for Train and Test

In [46]:
ypred_train = model.predict(xtrain)
ypred_test = model.predict(xtest)

In [47]:
ypred_train[0:5]

array([[2920.],
       [2530.],
       [3730.],
       [3490.],
       [4055.]])

In [48]:
ytrain.head()

Weight
57    2920
31    2530
62    3730
29    3490
51    4055

In [49]:
ypred_test[0:5]

array([[2573.61197373],
       [3617.27175293],
       [3040.87432563],
       [3332.77898679],
       [3970.06162749]])

In [50]:
ytest.head()

Weight
23    2670
86    3785
91    2985
21    3570
17    3910